src/unix/pipe.c

```diff
@@ -287,6 +287,67 @@ int uv_pipe_getpeername(const uv_pipe_t* handle, char* buffer, size_t* size) {
 }
 
 
+int uv_pipe_get_creds(const uv_pipe_t* handle, uv_pipe_creds_t* creds) {
+  int r;
+  uv_os_fd_t os_fd;
+
+  r = uv_fileno((uv_handle_t*) handle, &os_fd);
+  if (r == UV_EBADF)
+    return UV_EBADF;
+
+#if defined(__APPLE__) || defined(BSD)
```

Comment:
Does `defined(BSD)` actually work? We do explicit checks for `__FreeBSD__`, `__OpenBSD__`, etc. everywhere else.

Reply:
https://stackoverflow.com/a/26225829 According to that, yes. In fact, I believe libuv (or some associated part of the build system) defines `BSD` even for Darwin systems where clang doesn't do so by default (ran into that problem and had to invert some of the logic here).

Reply:
Right, only if `<sys/param.h>` is included - and that's done in `include/uv/unix.h`. You should probably update the comment in that file to say that it also pulls in the `BSD` macro.

Reply:
Good to know, will do

Reply:
@bnoordhuis let me know if that works.

Reply:
Did you perhaps forgot to push your changes? I'm not seeing them.

If you mean the comment you added here, I was suggesting you update the comment in _that_ file, `include/uv/unix.h`.

Reply:
Derp, gotcha. Done :)